pkg/gzip/gzip.go

```diff
@@ -165,3 +191,10 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 	}
 	return nil
 }
+
+func firstExistingParent(path string) string {
+	for _, err := os.Stat(path); err != nil; _, err = os.Stat(path) {
```

Comment:
I could envision a scenario where the filesystem is just unavailable somehow and all `stat` calls fail and this becomes an infinite loop